migration-util/src/main/java/edu/unc/lib/dcr/migration/TransformContentCommand.java

```diff
@@ -71,27 +74,39 @@
             description = "Submits the transformed content for deposit to the provided container UUID")
     private String depositInto;
 
+    @Option(names = {"-m", "--missing-deposit-records"},
+            description = "Create referenced deposit records which do not have their own bxc3 object")
+    private boolean createMissingDepositRecords;
+
+    @Option(names = {"-n", "--dry-run"},
+            description = "Perform the transformation but do not save the results")
+    private boolean dryRun;
+
+    private String applicationContextPath = "spring/service-context.xml";
+
     @Override
     public Integer call() throws Exception {
         long start = System.currentTimeMillis();
 
         output.info("Transforming content tree starting from {}", startingId);
         output.info("===========================================");
-        RepositoryPIDMinter pidMinter = new RepositoryPIDMinter();
+
+        ConfigurableApplicationContext context = new ClassPathXmlApplicationContext(applicationContextPath);
+
+        RepositoryPIDMinter pidMinter = (RepositoryPIDMinter) context.getBean("repositoryPIDMinter");
         PID depositPid = pidMinter.mintDepositRecordPid();
 
-        output.info("Populating deposit:");
-        output.info(depositPid.getId());
+        output.info("Populating deposit: " + depositPid.getId());
 
         DepositModelManager depositModelManager = new DepositModelManager(depositPid, parentCommand.tdbDir);
         DepositDirectoryManager depositDirectoryManager = new DepositDirectoryManager(
                 depositPid, parentCommand.depositBaseDir, hashNesting);
 
-        PathIndex pathIndex = new PathIndex();
+        PathIndex pathIndex = (PathIndex) context.getBean("pathIndex");
```

Comment:
Why are context files used?

Reply:
I converted this command over to using the spring context stuff because of adding in the fedora interactions, which required a whole bunch of extra dependencies and config options. The context stuff was already in use in the TransformDepositRecordCommand, so it was a pretty easy switch.